pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/readers/LazyRow.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.segment.readers;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.pinot.segment.spi.IndexSegment;
+
+
+/**
+ * <p>A wrapper class to read column values of a row for a given {@link IndexSegment} and docId.<br>
+ * The advantage of having wrapper over segment and docId is column values are read only when
+ * {@link LazyRow#getValue(String)} is invoked.
+ * This is useful to reduce the disk reads incurred due to loading the complete previous row during merge step.
+ *
+ * <p>The LazyRow has an internal state and should not be used concurrently. To reuse the LazyRow, create an instance
+ * using no arg constructor and re-initialise using {@link LazyRow#init(IndexSegment, int)}
+ */
+public class LazyRow {
+  private IndexSegment _segment;
+  private int _docId;
+  private final Map<String, Object> _fieldToValueMap = new HashMap<>();
+  private final Set<String> _nullValueFields = new HashSet<>();
+
+
+  public LazyRow() {
+  }
+
+  public LazyRow(IndexSegment segment, int docId) {
+    _segment = segment;
+    _docId = docId;
+  }
```

Comment:
Suggest removing them and only allow empty constructor. In order to use it, `init()` must be called first